### ISSUE TITLE
Integer gpgme versions are assumed, potentially leading to crash

### ISSUE THREAD

**Issue by satta** [gnome-keysign/gnome-keysign#93]
Debian unstable's gpgme (1.13.1) shows some interesting version information:
```
Python 3.7.4+ (default, Sep  4 2019, 08:03:05) 
[GCC 9.2.1 20190827] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gpg
>>> gpg.version.versionlist
['1', '13', '1-unknown']
```
This causes the following code in `keysign/gpgmeh.py` to crash, as it assumes that all components can be interpreted as integers:
```python
major, minor, patch = map(int, gpg.version.versionlist)
```
leading to test failures:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 417, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/build/gnome-keysign-1.0.1/tests/test_latin1.py", line 11, in <module>
    from keysign import gpgmeh
  File "/build/gnome-keysign-1.0.1/keysign/gpgmeh.py", line 38, in <module>
    major, minor, patch = map(int, gpg.version.versionlist)
ValueError: invalid literal for int() with base 10: '1-unknown'
```

Regardless of the format of the version string, I think that the code in gnome-keysign should be more robust when interpreting external strings.

**Comment by muelli** [comment]
good catch, thanks!

yeah, this has been a silly code path, anyway.
I am fine with assuming that the bug we worked around has been fixed on all our targetted platforms by now, so that we can get rid of that check.

**Comment by satta** [comment]
I have currently patched this on the Debian side with a small wrapper function catching this case, which I will remove as soon as there is a better solution in an upstream release. I'll keep an eye on new versions as usual -- thanks!

**Comment by muelli** [comment]
fixed with c7e524ba3e6dbf3c31d99197bcee50d51542a84f

**Comment by muelli** [comment]
JFTR: We're not alone in tripping over that version string: https://gitlab.gnome.org/GNOME/seahorse/merge_requests/92
